src/pages/Chat/steps/LivingSituationInput.tsx

```diff
@@ -58,15 +68,78 @@ const validationSchema = yup
       .oneOf(Object.keys(ApartmentType))
       .required(),
     size: yup
-      .number()
-      .moreThan(0)
-      .lessThan(250, 'TOO BIG APARTMENT') // TODO improve message
-      .required(),
+      .mixed()
+      .test({
+        name: 'isntTooBig',
+        test: (value) => !isNaN(Number(value)) && value < 250,
+        message: 'TOO BIG APARTMENT',
+      })
+      .test({
+        test: (value) =>
+          value === '' || (!isNaN(Number(value)) && Number(value) > 0),
+        message: 'MUST BE VALID NUMBER',
+      })
+      .test({
+        test: (value) => value !== '',
+        message: 'noop',
+      }),
   })
+
   .required()
 
-const isDone = (values: Partial<LivingSituationState> = {}) =>
-  validationSchema.isValidSync(values)
+const isDone = (values: Partial<LivingSituationState> = {}) => {
+  try {
+    validationSchema.validateSync(values)
+    return true
+  } catch (e) {
+    return false
+  }
+}
+const getValidationError = (
+  values: Partial<LivingSituationState> = {},
+): [string, string] | null => {
+  try {
+    validationSchema.validateSync(values)
+    return null
+  } catch (e) {
+    return [e.params.path, e.message]
+  }
+}
+
+const hasValidationErrorForKey = (key: keyof LivingSituationState) => (
+  validationError: yup.ValidationError,
+) =>
+  allPass([
+    pipe(
+      isNil,
+      not,
+    ),
+    pipe(
+      pathOr('', [0]),
+      equals(key),
+    ),
+    pipe(
+      pathOr('', [1]),
+      equals('noop'),
+      not,
+    ),
+  ])(validationError)
+
+const renderValidationError = (key: keyof LivingSituationState) => (
+  values: Partial<LivingSituationState>,
+): React.ReactNode =>
+  pipe(
```

Comment:
It was these kinds of things that I was afraid of when permitting ramda :/ I've spent like two minutes trying to decipher this and I've not still fully understand what really goes on in here 😅

Reply:
I guess we need to like... Make a decision whether we want to go all-in functional or not really then? coz it can totally be rewritten with if/else, variables and some duck type checking.

Reply:
yeah, I guess, let's discuss it :)

Reply:
Sounds good 👌

Reply:
I'm having some trouble reading this as well 🙁